main.cpp

```diff
@@ -115,17 +176,23 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
-    
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    T* tOne = new T( 33.3f, "tOne");                                             //6
+    T* tTwo = new T( 55.0f, "tTwo" );                                        //6
     
-    U <#name3#>;
+    Comparer f;
+    if( tOne != nullptr && tTwo != nullptr)                                            //7
+    {
+        auto* smaller = f.compare(tOne, tTwo );                              //8
+        std::cout << "the smaller one is << " << (smaller != nullptr ? smaller->name : "smaller is nullptr!") << std::endl; //9
+    }
+    else std::cout << "tOne and/or tTwo are nullptr!\n";
+    U* myUOne = new U();
```

Comment:
`new` has not been covered yet.
Declare normal instances on the stack.

You know how to take the address of variables now.  
use that knowledge to pass pointers to the functions that want pointers.
- [ ] revise